### PR TITLE
[BEAM-151] Move over some more Dataflow specific classes.

### PR DESCRIPTION
**Pull request by lukecwik**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

Note that users should use proto ByteString instead of RandomAccessData
since it provides a safer version of the same functionality.

I hoped that I would be able to move over more of the *Cloud* classes
and their helpers but they are embedded part of coders. Nothing more
can be done here until there is an official Beam representation of a coder
decoupled from Dataflow CloudKnownTypes.

### CODE REVIEW

**Comment by lukecwik** [comment]
R: @kennknowles

**Comment by kennknowles** [comment]
LGTM

**Comment by davorbonaci** [comment]
Can we coordinate with #239? Let's rename packages going into this module.

**Comment by kennknowles** [comment]
@davor you can pull these commits onto your PR

File: runners/google-cloud-dataflow-java/pom.xml [hunk 1]
```diff
@@ -524,6 +524,12 @@
       <version>1.0-rc2</version>
       <optional>true</optional>
     </dependency>
+    <dependency>
```
Comment (davorbonaci) [inline runners/google-cloud-dataflow-java/pom.xml]: empty line above

**Comment by davorbonaci** [comment]
LGTM

**Pull request closed without merging**